dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/analytics/dimension/DimensionFilteringAndPagingService.java

```diff
@@ -0,0 +1,158 @@
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.analytics.dimension;
+
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+
+import org.hisp.dhis.common.BaseIdentifiableObject;
+import org.hisp.dhis.common.DimensionsCriteria;
+import org.hisp.dhis.fieldfiltering.FieldFilterParams;
+import org.hisp.dhis.fieldfiltering.FieldFilterService;
+import org.hisp.dhis.webapi.controller.event.webrequest.OrderCriteria;
+import org.hisp.dhis.webapi.controller.event.webrequest.PagingAndSortingCriteriaAdapter;
+import org.hisp.dhis.webapi.controller.event.webrequest.PagingWrapper;
+import org.springframework.stereotype.Service;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.collect.ImmutableMap;
+
+@Service
+@RequiredArgsConstructor
+public class DimensionFilteringAndPagingService
+{
+
+    @NonNull
+    private final FieldFilterService fieldFilterService;
+
+    private final DimensionMapperService dimensionMapperService;
+
+    private final static Comparator<DimensionResponse> DEFAULT_COMPARATOR = Comparator
+        .comparing( DimensionResponse::getCreated );
+
+    private final static Map<String, Comparator<DimensionResponse>> ORDERING_MAP = ImmutableMap.of(
+        "lastUpdated", Comparator.comparing( DimensionResponse::getLastUpdated ),
+        "code", Comparator.comparing( DimensionResponse::getCode ),
+        "uid", Comparator.comparing( DimensionResponse::getId ),
+        "id", Comparator.comparing( DimensionResponse::getId ),
+        "name", Comparator.comparing( DimensionResponse::getName ) );
+
+    public PagingWrapper<ObjectNode> pageAndFilter(
+        Collection<BaseIdentifiableObject> dimensions,
+        DimensionsCriteria dimensionsCriteria,
+        List<String> fields )
+    {
+        Collection<DimensionResponse> dimensionResponses = dimensionMapperService.toDimensionResponse( dimensions );
+
+        PagingWrapper<ObjectNode> pagingWrapper = new PagingWrapper<>( "dimensions" );
+
+        Collection<DimensionResponse> filteredDimensions = filterStream( dimensionResponses.stream(),
+            dimensionsCriteria )
+                .collect( Collectors.toList() );
+
+        var params = FieldFilterParams.of( sortedAndPagedStream( filteredDimensions.stream(), dimensionsCriteria )
+            .collect( Collectors.toList() ), fields );
+
+        var objectNodes = fieldFilterService.toObjectNodes( params );
+
```

Comment:
Could we avoid using `var` unless there is a good reason for it? Makes it code less readable IMHO.